src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -58,12 +60,15 @@ ss::future<> refresh_credentials::do_start() {
     return ss::do_until(
       [this] { return _gate.is_closed() || _as.abort_requested(); },
       [this] {
-          return fetch_and_update_credentials().handle_exception_type(
-            [](const ss::sleep_aborted& ex) {
+          return fetch_and_update_credentials()
+            .handle_exception_type([](const ss::sleep_aborted& ex) {
                 vlog(
                   clrl_log.info,
                   "stopping refresh_credentials loop: {}",
                   ex.what());
+            })
+            .handle_exception([](const std::exception_ptr& ex) {
```

Comment:
With https://github.com/redpanda-data/redpanda/pull/8403 we should not end up here except for non-recoverable errors (such as misconfigured role etc). 

However retrying is safe even in such cases (although it may never succeed) so this change should be fine. 

I would suggest to increment failure count here so that we get increasing sleep interval between retries:

```suggestion
            .handle_exception([this](const std::exception_ptr& ex) {
                            _impl->increment_retries();
```

Reply:
Thanks, i will apply suggestion